src/LinuxTracing/StackSlice.h

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+#ifndef LINUX_TRACING_STACK_SLICE_H_
+#define LINUX_TRACING_STACK_SLICE_H_
+
+#include <absl/types/span.h>
+
+#include <memory>
+
+namespace orbit_linux_tracing {
+
+// This class holds a copy of some stack data from collected from the target process.
+// The class must have a single owner, to avoid unnecessary copies. Instead of passing pointers or
+// references of this class around, `StackSliceView` should be used instead.
+class StackSlice {
```

Comment:
Isn't this supposed to be used only in `UprobesUnwindingVisitor.cpp`? Do we need to have it in a header and expose it to the rest of the module? For example, `StackSliceView` doesn't really need to be coupled to this, it can just get address, size, data.

Reply:
This class needs to be known by the StackSliceView class, as it just makes sense to have StackSliceView to be constructed out of a StackSlice reference.

Also from their semantic meaning, they belong tightly together.

Reply:
> This class needs to be known by the StackSliceView class, as it just makes sense to have StackSliceView to be constructed out of a StackSlice reference.

Why? `StackSliceView` has to store an address, a size, a pointer. It could for example store a pointer to a stack directly in a `PerfEvent`.

> Also from their semantic meaning, they belong tightly together.

It's just a matter of how you look at it. For how I understand you are going to use them, there is no need to create a semantic coupling, they are just small utility classes to be used in two separate places.

Reply:
> It could for example store a pointer to a stack directly in a PerfEvent.

Actually, isn't that desired, so that we don't have to move from `StackSamplePerfEventData::data` into a `StackSlice`?